File: go/0455E-分发饼干.go
/*
 * @Author: SukiEva
 * @Date: 2021-11-15 10:22:48
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */
package main

import "sort"

func findContentChildren(g []int, s []int) int {
	sort.Ints(g)
	sort.Ints(s)
	j := 0
	for i := 0; i < len(s); i++ {
		if j < len(g) && g[j] <= s[i] {
			j++
		}
		if j >= len(g) {
			break
		}
	}
	return j
}

File: cpp/0455E-分发饼干.cpp
/*
 * @Author: SukiEva
 * @Date: 2021-12-30 19:12:25
 * @Description: https://github.com/SukiEva
 * わたし、気になります！
 */
#include <bits/stdc++.h>

#include "include/ListNode.h"
#include "include/TreeNode.h"

using namespace std;

class Solution {
   public:
    int findContentChildren(vector<int>& g, vector<int>& s) {
        sort(g.begin(), g.end());
        sort(s.begin(), s.end());
        int j = 0;
        for (int i = 0; i < s.size(); i++) {
            if (j < g.size() && s[i] >= g[j]) j++;
        }
        return j;
    }
};